aderyn_core/src/detect/nc/time_units_not_used.rs

```diff
@@ -0,0 +1,77 @@
+use std::{collections::BTreeMap, error::Error};
+
+use crate::{
+    ast::NodeID,
+    capture,
+    context::workspace_context::WorkspaceContext,
+    detect::detector::{IssueDetector, IssueDetectorNamePool, IssueSeverity},
+};
+use eyre::Result;
+
+#[derive(Default)]
+pub struct TimeUnitsNotUsedDetector {
+    // Keys are source file name and line number
+    found_instances: BTreeMap<(String, usize, String), NodeID>,
+}
+
+impl IssueDetector for TimeUnitsNotUsedDetector {
+    fn detect(&mut self, context: &WorkspaceContext) -> Result<bool, Box<dyn Error>> {
+        let disallow = [
+            "second", "seconds", "minute", "minutes", "hour", "hours", "day", "days", "week",
+            "weeks",
+        ];
+
+        for variable in context.variable_declarations() {
+            if disallow.iter().any(|&x| x == variable.name.to_lowercase()) {
```

Comment:
This feels too broad to me, and at the same time wouldn't catch many cases. This would catch:

```solidity
uint256 private constant DAY = 1 days;
```

but wouldn't catch:

```solidity
uint256 private constant 2_DAYS = 60*60*24*2;
```